tidy3d/components/simulation.py

```diff
@@ -908,6 +912,20 @@ def _validate_tfsf_nonuniform_grid(self) -> None:
                             f"axis, '{'xyz'[source.injection_axis]}'."
                         )
 
+    def _validate_customsourcetime_dt(self) -> None:
+        """Make sure custom source time is not undersampled."""
+        for source in self.sources:
+            if isinstance(source.source_time, CustomSourceTime):
+                dataset = source.source_time.source_time_dataset
+                if dataset is None:
+                    continue
+                times = dataset.values.coords["t"].values
+                max_dt = np.amax(np.diff(times))
+                if max_dt > self.dt / CUSTOMSOURCETIME_SAMPLING_WARNING_FACTOR:
+                    log.warning(
+                        "'CustomSourceTime' may be undersampled relative to 'Simulation.dt'"
```

Comment:
I'd just suggest making this warning a bit more descriptive. Printing the values of `max_dt` `self.dt ` and suggesting the solution, namely that we recommend the max dt of the custom source should be less than the dt of the simulation.